src/dsql/parse.y

```diff
@@ -2966,6 +2987,16 @@ package_clause
 		}
 	;
 
+%type <createAlterPackageNode> change_opt_package_clause
+change_opt_package_clause
```

Comment:
Maybe `change_opt_package_clause` and `change_opt_procedure_clause` could be put directly in the alter rules?
I think these rules has not so great names to be put standalone and they are not reused.

Reply:
I made rules `change_opt_package_clause` and `change_opt_procedure_clause` to separate full alter and partial alter. I can change the names to `partial_alter_package_clause` and `partial_alter_procedure_clause`. This way it will be clearer and will separate full alter and partial alter.

Reply:
You call it partial and I call it just another form of alter. Every alter is partial, they change what was specified. So my suggestion to just use another branch of the original rules directly.

Reply:
"Every alter is partial, they change what was specified." - is not correct for functions, procedures and packages. 
Functions/procedures/packages change all parameters, even those that were not specified. For example: 
```
ALTER FUNCTION func1 (str varchar(5))
RETURNS INT
AS
DECLARE result INTEGER;
BEGIN
result = POSITION('a', str);
RETURN result;
END;
```
The deterministic and sql security parameters were not specified, but are still being modified
(reset).
I will call this a full alter (the `executeAlter` method is used).

Now partial alter (`executeAlterIndividualParameters` method is used):
Here parameters that are not specified will not be modified.  Only the individual specified parameters will be changed.
For example:
```
ALTER FUNCTION name SQL SECURITY INVOKER
```
The deterministic parameter is not specified here and it is not modified.

My opinion:
This is different behavior and it should be in different branches.
But it is better to change the name to a more clarifying one: `partial_alter_package_clause` for example.

Reply:
So we have a big inconsistency with triggers:

```
SQL> create table t1 (n1 integer)!
SQL> create trigger t1_ins inactive before insert position 4 on t1 SQL SECURITY DEFINER as begin end!

SQL> show trigger t1_ins!
Triggers on Table T1:
T1_INS, Sequence: 4, Type: BEFORE INSERT, Inactive, SQL SECURITY DEFINER
Trigger text:
=============================================================================
as begin end
=============================================================================

SQL> alter trigger t1_ins as begin end!

SQL> show trigger t1_ins!
Triggers on Table T1:
T1_INS, Sequence: 4, Type: BEFORE INSERT, Inactive, SQL SECURITY DEFINER
Trigger text:
=============================================================================
as begin end
=============================================================================
```